### PR TITLE
Mod compositor async drawing

### PR DESCRIPTION
**Pull request by neithernut**
This PR introduces asynchronous drawing... via EGL.
Targets #418.

### CODE REVIEW

File: src/compositor/buffer/egl_target.c [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+egl_target_deinit(
+    struct ws_object* obj
+);
+
+/**
+ * Blit all the buffers
+ */
+static void
+blit_surfaces(
+    EV_P_
```
Comment (matthiasbeyer) [inline src/compositor/buffer/egl_target.c:50]: What is this? Can you document it please?

File: src/compositor/buffer/egl_target.c [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+int
+ws_egl_target_notify(
+    struct ws_egl_target* self
+) {
+    struct ev_loop* loop = ev_default_loop(EVFLAG_AUTO);
+    if (loop) {
+        return -1;
+    }
+
+    ev_async_send(loop, &self->watcher);
```
Comment (matthiasbeyer) [inline src/compositor/buffer/egl_target.c:162]: Do we need a write lock here? I'm not sure about this...

File: src/compositor/buffer/egl_target.c (hunk 1 above)
Comment (neithernut) [inline src/compositor/buffer/egl_target.c:50]: Look at the ev documentation.

File: src/compositor/buffer/egl_target.c (hunk 2 above)
Comment (neithernut) [inline src/compositor/buffer/egl_target.c:162]: Nope, it's thread safe.

**Comment by neithernut** [comment]
Mh... needs a major redesign...

**Pull request closed without merging**